server/storage.go

```diff
@@ -1,27 +1,65 @@
 package server
 
+import (
+	"fmt"
+)
+
 type User struct {
 	ID      string `json:"id"`
 	Name    string `json:"name"`
 	Surname string `json:"surname"`
 	Age     int    `json:"age"`
 }
 
-var storage []User
-
 type Storage interface {
 	GetUser(id string) (*User, error)
 	CreateUser(*User) error // Make sure ID is added to the struct OR return ID from this method
 	DeleteUser(id string) error
+	GetAllUsers() (*[]User, error)
+}
+
+type UserStorage struct {
+	storage []User
+}
+
+func (us *UserStorage) GetAllUsers() (*[]User, error) {
+	return &us.storage, nil
```

Comment:
We never use pointers to slices, slice is already sent by reference, not by value